demos/python_demos/whiteboard_inpainting_demo/utils/network_wrappers.py

```diff
@@ -0,0 +1,123 @@
+ limitations under the License.
+"""
+
+import json
+import logging as log
+
+import cv2
+import numpy as np
+
+from utils.ie_tools import load_ie_model
+from .segm_postprocess import postprocess
+
+
+class MaskRCNN:
+    """Wrapper class for a network returning masks of objects"""
+
+    def __init__(self, ie, model_path, conf=.6, device='CPU', ext_path='',
+                 max_reqs=100):
+        self.max_reqs = max_reqs
+        self.confidence = conf
+        self.net = load_ie_model(ie, model_path, device, None, ext_path, num_reqs=self.max_reqs)
+
+        required_input_keys = [{'im_info', 'im_data'}, {'im_data', 'im_info'}]
+        current_input_keys = set(self.net.inputs_info.keys())
+        assert current_input_keys in required_input_keys
+        required_output_keys = {'boxes', 'scores', 'classes', 'raw_masks'}
+        assert required_output_keys.issubset(self.net.net.outputs)
+
+        self.n, self.c, self.h, self.w = self.net.inputs_info['im_data'].shape
+        assert self.n == 1, 'Only batch 1 is supported.'
+
+    def preprocess(self, frame):
+        image_height, image_width = frame.shape[:2]
+        scale = min(self.h / image_height, self.w / image_width)
+        processed_image = cv2.resize(frame, None, fx=scale, fy=scale)
+        processed_image = processed_image.astype('float32').transpose(2, 0, 1)
+
+        sample = dict(original_image=frame,
+                      meta=dict(original_size=frame.shape[:2],
+                                processed_size=processed_image.shape[1:3]),
+                      im_data=processed_image,
+                      im_info=np.array([processed_image.shape[1], processed_image.shape[2], 1.0], dtype='float32'))
+        return sample
+
+    def forward(self, im_data, im_info):
+        if (self.h - im_data.shape[1] < 0) or (self.w - im_data.shape[2] < 0):
+            raise ValueError('Input image should resolution of {}x{} or less, '
+                             'got {}x{}.'.format(self.w, self.h, im_data.shape[2], im_data.shape[1]))
+        im_data = np.pad(im_data, ((0, 0),
+                                   (0, self.h - im_data.shape[1]),
+                                   (0, self.w - im_data.shape[2])),
+                         mode='constant', constant_values=0).reshape(1, self.c, self.h, self.w)
+        im_info = im_info.reshape(1, *im_info.shape)
+        output = self.net.net.infer(dict(im_data=im_data, im_info=im_info))
+
+        classes = output['classes']
+        valid_detections_mask = classes > 0
+        classes = classes[valid_detections_mask]
+        boxes = output['boxes'][valid_detections_mask]
+        scores = output['scores'][valid_detections_mask]
+        masks = output['raw_masks'][valid_detections_mask]
+        return boxes, classes, scores, np.full(len(classes), 0, dtype=np.int32), masks
+
+    def get_detections(self, frames, return_cropped_masks=False, only_class_person=True):
```

Comment:
`only_class_person` sounds strange.

Reply:
The argument was removed. Logic was reimplemented with list of named classes which will be hided.